### PR TITLE
Fix/add task perf

### PR DESCRIPTION
**Pull request by brnleehng**
- Removed upload blob function
- Shared args list
- Indexing through args

### CODE REVIEW

File: R/doAzureParallel.R [hunk 1]
```diff
@@ -548,8 +550,6 @@ setHttpTraffic <- function(value = FALSE) {
     return(taskId)
   })
 
-  rAzureBatch::updateJob(id)
-
   if (enableCloudCombine) {
     cat("\nSubmitting merge task")
     mergeTaskId <- paste0(id, "-merge")
```
Comment (paselem) [inline R/doAzureParallel.R]: just keep this as 'merge'? that way when searching we don't need to know the prefix by number.

File: R/doAzureParallel.R [hunk 2]
```diff
@@ -564,14 +564,16 @@ setHttpTraffic <- function(value = FALSE) {
...
-      dependsOn = tasks,
+      dependsOn = list(taskIdRanges = list(list(start = 1, end = length(tasks)))),
       cloudCombine = cloudCombine,
       outputFiles = obj$options$azure$outputFiles,
       containerImage = data$containerImage
     )
     cat(". . .")
   }
 
+  rAzureBatch::updateJob(id)
```
Comment (paselem) [inline R/doAzureParallel.R]: Why are you doing this? (add a comment). Also, I'm not sure I understand what is being updated? Do we need to pass in the job params?

File: R/helpers.R [hunk 3]
```diff
@@ -101,7 +83,7 @@
 
   commands <-
     c(commands,
-      dockerRunCommand(containerImage, rCommand, taskId))
+      dockerRunCommand(containerImage, rCommand, paste0("task-", taskId)))
```
Comment (paselem) [inline R/helpers.R]: do we still need to prepend "task-" to the command line here?

File: R/helpers.R [hunk 4]
```diff
@@ -34,22 +29,9 @@
     commands <- c(downloadCommand)
   }
 
-  envFile <- paste0(taskId, ".rds")
```
Comment (paselem) [inline R/helpers.R]: I think I might be missing something here, but don't we still need to upload a single argument blob to storage? Is that what was indended with the azbatchenv environment in the job prep? Are there any cases where this object gets too big  will not work? Would we ever need to chunk it up?

File: R/helpers.R (hunk 3 above)
Comment (brnleehng) [reply R/helpers.R]: Docker run has a character limit that requires the container name to be at least 3 characters or more

File: R/doAzureParallel.R (hunk 2 above)
Comment (brnleehng) [reply R/doAzureParallel.R]: We need to update the job property for terminating the job after all the tasks are completed

Comment (paselem) [reply R/doAzureParallel.R]: I think I get it, but the name of the function is quite confusing. Doesn't really say what is going on, and doesn't pass in any params.

File: R/helpers.R (hunk 3 above)
Comment (paselem) [reply R/helpers.R]: huh.. interesting. OK, seems fine then.

File: R/utility.R [hunk 5]
```diff
@@ -243,3 +243,15 @@ readMetadataBlob <- function(jobId) {
 areShallowEqual <- function(a, b) {
   !is.null(a) && !is.null(b) && a == b
 }
+
+hasDataSet <- function(list) {
+  if (length(list) > 0) {
+    for (arg in list[[1]]) {
+      if (typeof(arg) == "list") {
```
Comment (paselem) [inline R/utility.R]: Is 'list' the only type that would be a data set? Could it be something else like a data frame or something?

File: inst/startup/worker.R [hunk 6]
```diff
@@ -71,7 +72,24 @@ setwd(batchTaskWorkingDirectory)
 
 azbatchenv <-
   readRDS(paste0(batchJobPreparationDirectory, "/", batchJobEnvironment))
-argsList <- azbatchenv$argsList
+
+tryCatch({
+  print("Data Set Boolean: ")
```
Comment (paselem) [inline inst/startup/worker.R]: Kind of a hard to decipher print output. Can you make this (and lines 79 & 80) a bit more verbose?

File: inst/startup/worker.R [hunk 7]
```diff
@@ -71,7 +72,24 @@ setwd(batchTaskWorkingDirectory)
...
+
+  print(isDataSet)
+  print(typeof(isDataSet))
+  if (isDataSet) {
+    argsList <- readRDS(batchTaskEnvironment)
+  } else {
+    argsList <- azbatchenv$argsList
```
Comment (paselem) [inline inst/startup/worker.R]: Is this assignment needed? can't line 85 just be 
```R
argsList <- azbatchenv$argsList[startIndex:endIndex]
```

File: inst/startup/worker.R [hunk 8]
```diff
@@ -71,7 +72,24 @@ setwd(batchTaskWorkingDirectory)
...
+    argsList <- argsList[startIndex:endIndex]
+  }
+
+  print(argsList)
```
Comment (paselem) [inline inst/startup/worker.R]: nitpick: Add something before printing off argsList so we know what this is.

File: inst/startup/worker.R [hunk 9]
```diff
@@ -71,7 +72,24 @@ setwd(batchTaskWorkingDirectory)
...
 azbatchenv <-
   readRDS(paste0(batchJobPreparationDirectory, "/", batchJobEnvironment))
-argsList <- azbatchenv$argsList
+
+tryCatch({
+  print("Data Set Boolean: ")
+
+  print(isDataSet)
+  print(typeof(isDataSet))
+  if (isDataSet) {
```
Comment (paselem) [inline inst/startup/worker.R]: This really feels like something we can unit test pretty easily, hint :).

File: R/utility.R (hunk 5 above)
Comment (brnleehng) [reply R/utility.R]: Using the iterator as an argument, data sets will be a list

File: R/helpers.R (hunk 3 above)
Comment (brnleehng) [reply R/helpers.R]: Removed the name of the container

File: inst/startup/worker.R (hunk 9 above)
Comment (brnleehng) [reply inst/startup/worker.R]: Added a test for isDataSet

File: inst/startup/worker.R (hunk 8 above)
Comment (brnleehng) [reply inst/startup/worker.R]: This was for debugging purposes. Removed

**Pull request merged**